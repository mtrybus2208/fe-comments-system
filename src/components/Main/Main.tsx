import React from 'react';
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom';
import Login from '../../pages/Login//Login';
import Authentication from '../../components/Authentication/Authentication';
import PasswordReset from '../../pages/PasswordReset/PasswordReset';
import Snackbar from '../Snackbar/Snackbar';

const Main = () => {
  return (
    <Router>
      <Switch>
        <Route path="/" exact component={Login} />
        <Route path="/password-reset/:token" component={PasswordReset} />
        <Route path="*" component={Authentication} />
      </Switch>
      <Snackbar />
    </Router>
  );
};

export default Main;
