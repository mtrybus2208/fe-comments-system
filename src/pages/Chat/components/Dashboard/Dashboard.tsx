import React, { useEffect } from 'react';

import Users from '../Users/Users';

export interface DashboardProps {}

const Dashboard: React.FunctionComponent<DashboardProps> = () => {
  return (
    <>
      <p style={{ margin: '100px' }}>Dashboard</p>

      <Users />
    </>
  );
};

export default Dashboard;
