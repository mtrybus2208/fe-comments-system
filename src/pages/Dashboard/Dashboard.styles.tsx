import styled from 'styled-components';

export const Dashboard = styled.div`
  padding: 15px;
  margin-top: 50px;

  @media ${({ theme }) => theme.media.tablet} {
    padding: 30px;
    margin-top: 100px;
  }
`;
