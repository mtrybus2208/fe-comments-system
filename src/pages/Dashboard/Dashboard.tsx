import React from 'react';

export interface DashboardProps {}

const Dashboard: React.SFC<DashboardProps> = () => {
  return <p>Dashboard</p>;
};

export default Dashboard;
