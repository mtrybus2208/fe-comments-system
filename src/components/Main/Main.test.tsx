import React from 'React';
import { shallow } from 'enzyme';
import Main from './Main';
import { Route } from 'react-router';
import Dashboard from '../../pages/Dashboard/Dashboard';

describe('Main', () => {
  it('should render layout', () => {
    const wrapper = shallow(<Dashboard />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct dashboard route', () => {
    const wrapper = shallow(<Main />);
    const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.component;
      return pathMap;
    }, {});

    expect(pathMap['/dashboard']).toBe(Dashboard);
  });
});
