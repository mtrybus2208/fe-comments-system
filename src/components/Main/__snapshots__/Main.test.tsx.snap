// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render layout 1`] = `ShallowWrapper {}`;
